Issue 260 units

pisuke opened the pull request:
I've put together a pull request to include the BACnet units into the DBO list of units.
The reason for this is getting alignment between units in UDMI and DBO.
Let me know if you have any comments. 
@grafnu should this be successful, should we also change the `units.json` file in the UDMI schema to have the same snakecase way of writing units?
As a side note, there is a very complete units ontology: [QUDT](http://www.qudt.org/2.1/catalog/qudt-catalog.html). It would be good to do an alignment between DBO and QUDT.
Also, DBO units miss to define conversion factors between standard unit and alternative ones: is this something that will be added in the future?
Thanks for having a look!

grafnu commented on ontology/yaml/resources/units/units.yaml:
```diff
@@ -12,58 +12,319 @@
+-	degrees_phase: STANDARD
+
+electromagneticfieldstrength:
+-	amperes_per_meter: STANDARD
```
spacing seems inconsistent -- why is this line (72) different than say the next one (75)?

pisuke replied:
Thanks, it was an issue with tabs being kept after a copy/paste from the BACnet units specs. Corrected now, thanks for spotting it!

grafnu reviewed (commented):
RE: UDMI, once this is in, I likely think units.json will just go away and be replaced by a simple regex check on a valid unit structure (snake case).  No need/desire to explicitly list them there as well.

charbull reviewed (changes requested):
We are relying on QUDT already.
We are going through an internal clean up to update the units internally first then export them to github.
Ideally revert the deletions/renaming and keep only the additions as they will be easier to update internally.

Thank you

pisuke commented:
> RE: UDMI, once this is in, I likely think units.json will just go away and be replaced by a simple regex check on a valid unit structure (snake case). No need/desire to explicitly list them there as well.

makes perfect sense, thanks for clarifying

pisuke commented:
> We are relying on QUDT already.
> We are going through an internal clean up to update the units internally first then export them to github.
> Ideally revert the deletions/renaming and keep only the additions as they will be easier to update internally.
> 
> Thank you

@charbull I've addressed this comment and reverted the deletions and renaming, so there should be just additions now.
 I don't agree with some of the quantity names (level vs length, I've decided to keep both), specificenthalpy could just be enthalpy and some others, but we could discuss this once you have done your internal units update.

Any reason why the quantities don't keep underscore characters to separate words? They are a bit difficult to read.

charbull commented:
@pisuke can you fix the validation errors here please: https://github.com/google/digitalbuildings/pull/276/checks?check_run_id=3021112121

charbull reviewed (changes requested):
@pisuke  please fix the errors in the validation : https://github.com/google/digitalbuildings/pull/276/checks?check_run_id=3021112121

pisuke commented:
> @pisuke please fix the errors in the validation : https://github.com/google/digitalbuildings/pull/276/checks?check_run_id=3021112121

@charbull OK, I've added the quantities descriptions as requested by the validation process and the validation checks now pass.
However, I disagree with forcing singularity of unit, and for this reason I've commented the "duplicates" I would like to suggest to introduce. 
For instance, meters at the moment a unit for "level" but I think it should also be recorded as a unit for "length".
"no_unit" is a special case of adimensionality to measure powerfactor, but it will be handy for any adimensional quantity.
Thoughts?

charbull commented:
@tsodorff can you please take a look at this?

tasodorff commented on ontology/yaml/resources/subfields/subfields.yaml:
```diff
@@ -173,22 +173,63 @@ descriptor:
+  electricfield: "A region around a charged particle or object within which a force would be exerted on other charged particles or objects."
+  electricinductance: "The property of an electric conductor or circuit that causes an electromotive force to be generated by a change in the current flowing."
+  electricphasedifference: "The time interval by which a wave leads by or lags by another wave."
+  electromagneticfieldstrength: "Quantitative expression of the intensity of an electric field at a particular location"
```
is this an electric field or a magnetic field? they measure different things; electric field is in V/m and magnetic field is in A/m.

pisuke replied:
updated to electric

tasodorff commented on ontology/yaml/resources/subfields/subfields.yaml:
```diff
@@ -173,22 +173,63 @@ descriptor:
+  electricinductance: "The property of an electric conductor or circuit that causes an electromotive force to be generated by a change in the current flowing."
+  electricphasedifference: "The time interval by which a wave leads by or lags by another wave."
+  electromagneticfieldstrength: "Quantitative expression of the intensity of an electric field at a particular location"
+  electromagneticmoment: "Measure of the magnetic strength of a magnet or current-carrying coil, expressed as the torque produced when the magnet or coil is set with its axis perpendicular to unit magnetic flux density."
```
magnetic, not electromagnetic

pisuke replied:
updated to magnetic

tasodorff reviewed (changes requested):
Some comments on differentiating between magnetic and electric in electromagnetic keywords. Please be clear about if things are electric or magnetic in nature (e.g. which type of field it is).

biotvandenberg commented on ontology/yaml/resources/units/units.yaml:
```diff
@@ -12,58 +12,315 @@
 - joules: STANDARD
 - kilowatt_hours
+- btus
+- kilobtus
```
Should the kilo* and mega* units be consistent with underscores?  e.g. this has kilobtus and mega_btus.
Should it be one of:
- kilo_btus and mega_btus or
- kilobtus and megabtus

pisuke replied:
thanks for spotting this, for consistency all metric prefixes have been replaced without the underscore

pisuke commented on ontology/yaml/resources/subfields/subfields.yaml:
```diff
@@ -173,22 +173,63 @@ descriptor:
+  electriccurrentdensity: "The amount of electric current flowing per unit cross-sectional area of a material."
+  electricfield: "A region around a charged particle or object within which a force would be exerted on other charged particles or objects."
+  electricinductance: "The property of an electric conductor or circuit that causes an electromotive force to be generated by a change in the current flowing."
+  electricphasedifference: "The time interval by which a wave leads by or lags by another wave."
```
thanks for the suggestion, description updated to "angular difference"

pisuke commented:
I've addressed all the suggestions, it should be ready to be merged.
For info, another good source of units and quantities information is this https://github.com/ryantenney/gnu-units/blob/master/units.dat
It also includes conversion factors.

pisuke commented:
@charbull I think I've addressed all the requests, can this PR now be merged?

charbull reviewed (changes requested):
@pisuke can you update to resolve the conflicts please?

mschulze17 commented:
#414 has been merged. @charbull, @pisuke - can this PR be addressed now?
